scripts/globals/fishing_contest.lua

```diff
@@ -0,0 +1,417 @@
+                    [1] = contest['fishid'],
+                    [2] = contest['criteria'],
+                    [3] = contest['measure'],
+                })
+            elseif contest['status'] == xi.fishingcontest.status.PRESENTING then
+                player:updateEvent({
+                    [0] = contest['status'],
+                    [1] = contest['fishid'],
+                    [2] = contest['criteria'],
+                    [3] = contest['measure'],
+                    [4] = isRewardAvailable(player),
+                })
+            end
+        elseif option == 150 then
+        -- Initial option
+            local time = getTimeRemaining(contest['changetime'])
+            if contest['status'] == xi.fishingcontest.status.ACCEPTING then
+                player:updateEvent({
+                    [0] = contest['status'],
+                    [1] = contest['fishid'],
+                    [2] = contest['criteria'],
+                    [3] = contest['measure'],
+                    [4] = isRewardAvailable(player),
+                    [5] = time['days'],
+                    [6] = time['hours'],
+                    [7] = time['minutes'],
+                })
+            elseif contest['status'] == xi.fishingcontest.status.PRESENTING then
+                player:updateEvent({
+                    [0] = contest['status'],
+                    [1] = contest['fishid'],
+                    [2] = contest['criteria'],
+                    [3] = contest['measure'],
+                    [4] = isRewardAvailable(player),
+                    [5] = time['days'],
+                    [6] = time['hours'],
+                    [7] = time['minutes'],
+                })
+            end
+        elseif option == 151 then
+        -- Confirm request to confirm entry
+            player:updateEvent({
+                [0] = contest['status'],
+                [1] = contest['fishid'],
+                [2] = contest['criteria'],
+                [3] = contest['measure'],
+                [5] = player:getContestScore(),
+            })
+        elseif option == 152 then
+        -- Confirm request to withdraw entry
+            player:withdrawContestFish()
+        end
+
+    elseif csid == 10007 then
+        if option == 144 then
+            if player:getGil() < 500 then
+                -- Player does not have enough gil
+                player:messageSpecial(zones[player:getZoneID()].text.INSUFFICIENT_GIL)
+                return
+            else
+                player:updateEvent()
+            end
+        elseif option == 145 then
+            player:updateEvent()
+        end
+    end
+end
+
+xi.fishingcontest.onEventFinish = function(player, csid, option, npc)
+    local contest = GetFishingContest()
+
+    if csid == 10007 then
+        -- Give the player back a fish with the same stats, but tagged for Ranking
+        if option == 145 then
+            local length = player:getLocalVar('[FishContest]Length')
```

Comment:
Is this needed on either side of the trade? Or is this a thing that can be lost if the player zones?

Reply:
There shouldn't be any zoning since this is during the interaction with the NPC.  The local var is just a temporary holder for the fish stats so that they can receive the same fish back but marked for ranking (i.e. so they cannot use it a second time).